docs/resources/artifactory_xray_watch.md

```diff
@@ -1,64 +1,187 @@
-    name = xray_policy.example.name
-    type = "license"
+	
+  repository_paths {
+    include_patterns = [
+      "path1/**",
+      "another-path/**",
+    ]
+
+    exclude_patterns = [
+      "path1/ignore/**",
+      "another-path/ignore**",
+    ]
   }
 }
 ```
 
-## Argument Reference
-
-The following arguments are supported:
+### Example Watch for named repositories
+```hcl
+resource "artifactory_local_repository" "example" {
+  key 	       = "local-repo"
+  package_type = "generic"
+  xray_index   = true
+}
 
-* `name` - (Required) Name of the watch (must be unique)
-* `description` - (Optional) Description of the watch
-* `active` - (Optional) Whether or not the watch will be active
-* `resources` - (Required) Nested argument describing the resources to be watched. Defined below.
-* `assigned_policies` - (Required) Nested argument describing policies that will be applied. Defined below.
+resource "artifactory_xray_watch" "watch" {
+  name        = "named-local-repo"
+  description = "all repositories"
+
+  repository {
+    name = artifactory_local_repository.example.key
+
+    package_types = ["Generic"]
+    paths         = ["path/*"]
+    mime_types    = ["application/zip"]
+
+    property {
+      key   = "field1"
+      value = "value 1"
+    }
+    property {
+      key   = "field2"
+      value = "value 2"
+    }
+  }
 
-### resources
+  repository_paths {
+    include_patterns = [
+      "path1/**",
+    ]
+    exclude_patterns = [
+      "path1/ignore/**",
+    ]
+  }
+}
+```
 
-The top-level `resources` block contains a list of one or more resource objects that each support the following:
+### Example Watch for all builds
+```hcl
+resource "artifactory_xray_watch" "all_builds" {
+  name        = "all_builds"
+  description = "all builds"
+  active      = true
 
-* `type` - (Required) Type of resource to be watched
-* `name` - (Required) A name describing the resource
-* `bin_mgr_id` - (Optional) The ID number of a binary manager resource
-* `repo_type` - (Optional) Type of repository (e.g. local or remote)
-* `filters` - (Optional) Nested argument describing filters to be applied. Defined below.
+  all_builds {
+    bin_mgr_id = "default"
+  }
 
-#### filters
+  policy {
+    name = "policy-1"
+    type = "security"
+  }
+}
+```
 
-The nested `filters` block contains a list of one or more filters to be applied, each of which supports the following:
+### Example Watch for all builds, filtered by pattern
+```hcl
+resource "artifactory_xray_watch" "pattern_builds" {
+  name        = "pattern_builds"
+  description = "builds by pattern"
+  active      = true
+
+  all_builds {
+    include_patterns = ["hello/**", "apache/**"]
+    exclude_patterns = ["apache/bad*", "world/**"]
+    bin_mgr_id       = "default"        
+  }
 
-* `type` - (Required) The type of filter, such as `regex` or `package-type`
-* `value` - (Required) The value of the filter, such as the text of the regex or name of the package type
+  policy {
```

Comment:
The idea here is that it creates the policy right here? Policies are independent resources - If the policy is created elsewhere you don't need to specify the type

Reply:
Ok, I went and looked up the xray client code on this and it just shoves this over-the-wire. What happens when you try to give it a non-existent policy name or a policy name that's correct but the wrong type?

It would be good if the validation code actually went and fetched the policies so it can be validated at plan time

Reply:
When running `terraform apply` and the watch **does not** exist, the api responds with:
```bash
Error: Xray response: 400 Bad Request
{
  "error": "Got invalid watch: policy blah doen't exist"
}
```

When running `terraform apply` and the watch **does exist** but has a **wrong type**, the api responds with:
```bash
Error: Xray response: 400 Bad Request
{
  "error": "Got invalid watch: policy badpolicytype type should be security"
}
```

Other resources don't lookup dependent resources during a plan. For example, virtual repos don't look up local repos.
I'm not sure if it's necessary to implement.

Reply:
I guess it's a question of fail fast or fail later. In this case, fail fast requires an API call first. I guess it's probably ok to fail later

Reply:
Hmm.. Let's consider a different scenario: Both the policy and watch are created in the same terraform code.

If we implement the lookup of the `policy` in `artifactory_xray_watch`, it will always fail because it doesn't exist.

```hcl
resource "artifactory_xray_policy" "test" {
  name  = "all-repositories"
  description = "watch-test"
  type = "security"

  rules {
    name = "rule1"
    priority = 1
    criteria {
      min_severity = "High"
    }
    actions {
      block_download {
        unscanned = true
        active = true
      }
    }
  }
}

resource "artifactory_xray_watch" "watch" {
  name        = "all-repositories"
  description = "all repositories"

  all_repositories {}

  policy {
    name = artifactory_xray_policy.test.id
    type = "security"
  }
}
```

Reply:
Yeah, I simply don't like our apis (don't tell anyone I said that :shh:)

Reply:
can you please call out these limitations in the docs?